src/modules/lua.c

```diff
@@ -183,11 +189,10 @@ mtev_lua_lmc_setL(lua_module_closure_t *lmc, lua_State *L) {
 
 void
 mtev_lua_cancel_coro(mtev_lua_resume_info_t *ci) {
-  lua_getglobal(ci->lmc->lua_state, "mtev_coros");
-  luaL_unref(ci->lmc->lua_state, -1, ci->coro_state_ref);
-  lua_pop(ci->lmc->lua_state, 1);
-  lua_gc(ci->lmc->lua_state, LUA_GCCOLLECT, 0);
   mtevL(nldeb, "coro_store <- %p\n", ci->coro_state);
+  luaL_unref(ci->lmc->lua_state, LUA_REGISTRYINDEX, ci->coro_state_ref);
+  lua_pop(ci->lmc->lua_state, 1);
+  mtev_lua_gc(ci->lmc);
```

Comment:
I think we always want to do a full collection here ?
```suggestion
  lua_gc(ci->lmc->lua_state, LUA_GCCOLLECT, 0);
```

Reply:
This would move it back inline, I added a mtev_lua_gc_full function that will force a full on next tick.